nuxeo-ai-core/src/main/java/org/nuxeo/ai/auto/AutoService.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Gethin James
+ */
+package org.nuxeo.ai.auto;
+
+import org.nuxeo.ecm.core.api.DocumentModel;
+
+/**
+ * Autofill and AutoCorrect services
+ */
+public interface AutoService {
+
+    /**
+     * Calculate all properties on a Document.
+     */
+    void calculateProperties(DocumentModel doc);
+
+    /**
+     * Calculate all properties on a Document, given the type of calculation.
+     */
+    void calculateProperties(DocumentModel doc, AUTO_ACTION action);
+
+    /**
+     * Remove the property from the list of auto updated values and remove suggestions.
+     */
+    void confirmProperty(DocumentModel doc, String xpath);
```

Comment:
Is this the best name ? I am terrible naming, but took me a while to get what it is.
How about updatedByHuman() or confirmedByHuman() confirmedByOperator()

Reply:
I agree, its not actually implemented yet so I will change the name when we do the implementation.